let ValidationError,DEBUG_MODE,CustomErrorHandler;_f36‍.x([["default",()=>_f36‍.o]]);_f36‍.w("joi",[["ValidationError",["ValidationError"],function(v){ValidationError=v}]]);_f36‍.w("../config/index.js",[["DEBUG_MODE",["DEBUG_MODE"],function(v){DEBUG_MODE=v}]]);_f36‍.w("../Services/CustomErrorHandler.js",[["default",["CustomErrorHandler"],function(v){CustomErrorHandler=v}]]);




const errorHandler = (err, req, res, next) => {
    let statusCode = 500;
    
    let data = {
        message: 'Internal server error',
        ...(DEBUG_MODE === 'true' && { originalError: err.message })
        
    }
    if(err instanceof ValidationError){
        statusCode = 422;
        data = {
            message: err.message
        }
    }
    if( err instanceof CustomErrorHandler){
        statusCode = err.status;
        data = {
            message: err.message
        }

    }
    return res.status(statusCode).json(data);


}

_f36‍.d(errorHandler);